.github/workflows/main.yaml

```diff
@@ -0,0 +1,170 @@
+# Reusable Cobalt CI workflow.
+
+name: main
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        description: 'Cobalt platform.'
+        required: true
+        type: string
+
+# Global env vars.
+env:
+  REGISTRY: ghcr.io
+  IPV6_AVAILABLE: 0
+  IS_BUILDBOT_DOCKER: 1
+  #BUILD_ID_SERVER_URL:
+  IS_CI: 1
+  IS_DOCKER: 1
+  NINJA_STATUS: '[%e sec | %f/%t %u remaining | %c/sec | j%r]'
+  SCCACHE: 1
+  SCCACHE_GCS_BUCKET: cobalt-actions-devel-sccache-linux
+  SCCACHE_GCS_OAUTH_URL: http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+  SCCACHE_GCS_RW_MODE: READ_WRITE
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Builds, tags, and pushes Cobalt docker images to ghr.
+  docker-image:
```

Comment:
Does the docker-image job need to be in the same file as the build-matrix + build jobs? It'd be nice if we could split these jobs out, but I'm not sure if we run into the yaml limitations you were talking about

Reply:
Chatted with Andrew offline, we are planning to move some of these steps to separate actions.

Reply:
Moved docker image build to a separate action.